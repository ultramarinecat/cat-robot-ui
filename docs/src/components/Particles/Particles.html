<!DOCTYPE html>

<html>
<head>
  <title>Particles.js</title>
  <meta http-equiv="content-type" content="text/html; charset=UTF-8">
  <meta name="viewport" content="width=device-width, target-densitydpi=160dpi, initial-scale=1.0; maximum-scale=1.0; user-scalable=0;">
  <link rel="stylesheet" media="all" href="../../../docco.css" />
</head>
<body>
  <div id="container">
    <div id="background"></div>
    
      <ul id="jump_to">
        <li>
          <a class="large" href="javascript:void(0);">Jump To &hellip;</a>
          <a class="small" href="javascript:void(0);">+</a>
          <div id="jump_wrapper">
          <div id="jump_page_wrapper">
            <div id="jump_page">
              
                
                <a class="source" href="../App/App.html">
                  src/components/App/App.js
                </a>
              
                
                <a class="source" href="../App/ErrorBoundary.html">
                  src/components/App/ErrorBoundary.js
                </a>
              
                
                <a class="source" href="../App/i18n.html">
                  src/components/App/i18n.js
                </a>
              
                
                <a class="source" href="../Auth/Auth.html">
                  src/components/Auth/Auth.js
                </a>
              
                
                <a class="source" href="../Auth/AuthActions.html">
                  src/components/Auth/AuthActions.js
                </a>
              
                
                <a class="source" href="../Auth/AuthReducer.html">
                  src/components/Auth/AuthReducer.js
                </a>
              
                
                <a class="source" href="../Button/Button.html">
                  src/components/Button/Button.js
                </a>
              
                
                <a class="source" href="../Cat/Cat.html">
                  src/components/Cat/Cat.js
                </a>
              
                
                <a class="source" href="../Cat/CatActions.html">
                  src/components/Cat/CatActions.js
                </a>
              
                
                <a class="source" href="../Cat/CatReducer.html">
                  src/components/Cat/CatReducer.js
                </a>
              
                
                <a class="source" href="../Cat/fulltilt.html">
                  src/components/Cat/fulltilt.js
                </a>
              
                
                <a class="source" href="../Connection/Connection.html">
                  src/components/Connection/Connection.js
                </a>
              
                
                <a class="source" href="../Connection/ConnectionActions.html">
                  src/components/Connection/ConnectionActions.js
                </a>
              
                
                <a class="source" href="../Connection/ConnectionReducer.html">
                  src/components/Connection/ConnectionReducer.js
                </a>
              
                
                <a class="source" href="../Connection/messages.html">
                  src/components/Connection/messages.js
                </a>
              
                
                <a class="source" href="../Intro/Intro.html">
                  src/components/Intro/Intro.js
                </a>
              
                
                <a class="source" href="../Intro/IntroActions.html">
                  src/components/Intro/IntroActions.js
                </a>
              
                
                <a class="source" href="Particles.html">
                  src/components/Particles/Particles.js
                </a>
              
                
                <a class="source" href="../Pulse/Pulse.html">
                  src/components/Pulse/Pulse.js
                </a>
              
                
                <a class="source" href="../Router/Router.html">
                  src/components/Router/Router.js
                </a>
              
                
                <a class="source" href="../Sound/CatSound.html">
                  src/components/Sound/CatSound.js
                </a>
              
                
                <a class="source" href="../Sound/Mute.html">
                  src/components/Sound/Mute.js
                </a>
              
                
                <a class="source" href="../Sound/Sound.html">
                  src/components/Sound/Sound.js
                </a>
              
                
                <a class="source" href="../Sound/TurningSound.html">
                  src/components/Sound/TurningSound.js
                </a>
              
                
                <a class="source" href="../Spinner/Spinner.html">
                  src/components/Spinner/Spinner.js
                </a>
              
                
                <a class="source" href="../Typing/Typing.html">
                  src/components/Typing/Typing.js
                </a>
              
                
                <a class="source" href="../../configureStore.html">
                  src/configureStore.js
                </a>
              
                
                <a class="source" href="../../index.html">
                  src/index.js
                </a>
              
                
                <a class="source" href="../../serviceWorker.html">
                  src/serviceWorker.js
                </a>
              
                
                <a class="source" href="../../setupTests.html">
                  src/setupTests.js
                </a>
              
            </div>
          </div>
        </li>
      </ul>
    
    <ul class="sections">
        
          <li id="title">
              <div class="annotation">
                  <h1>Particles.js</h1>
              </div>
          </li>
        
        
        
        <li id="section-1">
            <div class="annotation">
              
              <div class="pilwrap ">
                <a class="pilcrow" href="#section-1">&#182;</a>
              </div>
              <p>Particles
Displays snow particles effect</p>

            </div>
            
            <div class="content"><div class='highlight'><pre><span class="hljs-comment">/* eslint-disable quotes */</span>

<span class="hljs-keyword">import</span> React, { memo, useEffect } <span class="hljs-keyword">from</span> <span class="hljs-string">'react'</span>;

<span class="hljs-keyword">import</span> <span class="hljs-string">'particles.js'</span>;
<span class="hljs-keyword">import</span> styles <span class="hljs-keyword">from</span> <span class="hljs-string">'./Particles.module.scss'</span>;

<span class="hljs-keyword">const</span> PARTICLES_ID = <span class="hljs-string">'particles'</span>;

<span class="hljs-keyword">export</span> <span class="hljs-function"><span class="hljs-keyword">function</span> <span class="hljs-title">Particles</span>(<span class="hljs-params"></span>) </span>{
  useEffect(<span class="hljs-function"><span class="hljs-params">()</span> =&gt;</span> {
    initParticles();
  }, []);

  <span class="hljs-keyword">return</span> <span class="xml"><span class="hljs-tag">&lt;<span class="hljs-name">div</span> <span class="hljs-attr">id</span>=<span class="hljs-string">{PARTICLES_ID}</span> <span class="hljs-attr">className</span>=<span class="hljs-string">{styles.main}</span> <span class="hljs-attr">data-testid</span>=<span class="hljs-string">"particles"</span> /&gt;</span>;
}

function initParticles() {
</span></pre></div></div>
            
        </li>
        
        
        <li id="section-2">
            <div class="annotation">
              
              <div class="pilwrap ">
                <a class="pilcrow" href="#section-2">&#182;</a>
              </div>
              <p><a href="https://github.com/VincentGarreau/particles.js/#options">https://github.com/VincentGarreau/particles.js/#options</a>
prettier-ignore</p>

            </div>
            
            <div class="content"><div class='highlight'><pre>  particlesJS(PARTICLES_ID, { <span class="hljs-comment">// eslint-disable-line no-undef</span>
    <span class="hljs-string">"particles"</span>: {
      <span class="hljs-string">"number"</span>: {</pre></div></div>
            
        </li>
        
        
        <li id="section-3">
            <div class="annotation">
              
              <div class="pilwrap ">
                <a class="pilcrow" href="#section-3">&#182;</a>
              </div>
              <p>“value”: 180,</p>

            </div>
            
            <div class="content"><div class='highlight'><pre>        <span class="hljs-string">"value"</span>: <span class="hljs-number">260</span>,
        <span class="hljs-string">"density"</span>: {
          <span class="hljs-string">"enable"</span>: <span class="hljs-literal">true</span>,
          <span class="hljs-string">"value_area"</span>: <span class="hljs-number">800</span>
        }
      },
      <span class="hljs-string">"color"</span>: {
        <span class="hljs-string">"value"</span>: <span class="hljs-string">"#fff"</span>
      },
      <span class="hljs-string">"shape"</span>: {
        <span class="hljs-string">"type"</span>: <span class="hljs-string">"circle"</span>,
        <span class="hljs-string">"stroke"</span>: {
          <span class="hljs-string">"width"</span>: <span class="hljs-number">0</span>,
          <span class="hljs-string">"color"</span>: <span class="hljs-string">"#000000"</span>
        },
        <span class="hljs-string">"polygon"</span>: {
          <span class="hljs-string">"nb_sides"</span>: <span class="hljs-number">5</span>
        },
        <span class="hljs-string">"image"</span>: {
          <span class="hljs-string">"src"</span>: <span class="hljs-string">"img/github.svg"</span>,
          <span class="hljs-string">"width"</span>: <span class="hljs-number">100</span>,
          <span class="hljs-string">"height"</span>: <span class="hljs-number">100</span>
        }
      },
      <span class="hljs-string">"opacity"</span>: {
        <span class="hljs-string">"value"</span>: <span class="hljs-number">0.5</span>,
        <span class="hljs-string">"random"</span>: <span class="hljs-literal">true</span>,
        <span class="hljs-string">"anim"</span>: {
          <span class="hljs-string">"enable"</span>: <span class="hljs-literal">false</span>,
          <span class="hljs-string">"speed"</span>: <span class="hljs-number">1</span>,
          <span class="hljs-string">"opacity_min"</span>: <span class="hljs-number">0.1</span>,
          <span class="hljs-string">"sync"</span>: <span class="hljs-literal">false</span>
        }
      },
      <span class="hljs-string">"size"</span>: {
        <span class="hljs-string">"value"</span>: <span class="hljs-number">3.2</span>,
        <span class="hljs-string">"random"</span>: <span class="hljs-literal">true</span>,
        <span class="hljs-string">"anim"</span>: {
          <span class="hljs-string">"enable"</span>: <span class="hljs-literal">false</span>,
          <span class="hljs-string">"speed"</span>: <span class="hljs-number">40</span>,
          <span class="hljs-string">"size_min"</span>: <span class="hljs-number">0.1</span>,
          <span class="hljs-string">"sync"</span>: <span class="hljs-literal">false</span>
        }
      },
      <span class="hljs-string">"line_linked"</span>: {
        <span class="hljs-string">"enable"</span>: <span class="hljs-literal">false</span>,
        <span class="hljs-string">"distance"</span>: <span class="hljs-number">500</span>,
        <span class="hljs-string">"color"</span>: <span class="hljs-string">"#ffffff"</span>,
        <span class="hljs-string">"opacity"</span>: <span class="hljs-number">0.4</span>,
        <span class="hljs-string">"width"</span>: <span class="hljs-number">2</span>
      },
      <span class="hljs-string">"move"</span>: {
        <span class="hljs-string">"enable"</span>: <span class="hljs-literal">true</span>,</pre></div></div>
            
        </li>
        
        
        <li id="section-4">
            <div class="annotation">
              
              <div class="pilwrap ">
                <a class="pilcrow" href="#section-4">&#182;</a>
              </div>
              <p>“speed”: 0.8,
“speed”: 0.55,</p>

            </div>
            
            <div class="content"><div class='highlight'><pre>        <span class="hljs-string">"speed"</span>: <span class="hljs-number">0.65</span>,
        <span class="hljs-string">"direction"</span>: <span class="hljs-string">"bottom"</span>,
        <span class="hljs-string">"random"</span>: <span class="hljs-literal">false</span>,
        <span class="hljs-string">"straight"</span>: <span class="hljs-literal">false</span>,
        <span class="hljs-string">"out_mode"</span>: <span class="hljs-string">"out"</span>,
        <span class="hljs-string">"bounce"</span>: <span class="hljs-literal">false</span>,
        <span class="hljs-string">"attract"</span>: {
          <span class="hljs-string">"enable"</span>: <span class="hljs-literal">false</span>,
          <span class="hljs-string">"rotateX"</span>: <span class="hljs-number">600</span>,
          <span class="hljs-string">"rotateY"</span>: <span class="hljs-number">1200</span>
        }
      }
    },
    <span class="hljs-string">"interactivity"</span>: {
      <span class="hljs-string">"detect_on"</span>: <span class="hljs-string">"canvas"</span>,
      <span class="hljs-string">"events"</span>: {
        <span class="hljs-string">"onhover"</span>: {
          <span class="hljs-string">"enable"</span>: <span class="hljs-literal">false</span>,
          <span class="hljs-string">"mode"</span>: <span class="hljs-string">"bubble"</span>
        },
        <span class="hljs-string">"onclick"</span>: {
          <span class="hljs-string">"enable"</span>: <span class="hljs-literal">true</span>,
          <span class="hljs-string">"mode"</span>: <span class="hljs-string">"repulse"</span>
        },
        <span class="hljs-string">"resize"</span>: <span class="hljs-literal">true</span>
      },
      <span class="hljs-string">"modes"</span>: {
        <span class="hljs-string">"grab"</span>: {
          <span class="hljs-string">"distance"</span>: <span class="hljs-number">400</span>,
          <span class="hljs-string">"line_linked"</span>: {
            <span class="hljs-string">"opacity"</span>: <span class="hljs-number">0.5</span>
          }
        },
        <span class="hljs-string">"bubble"</span>: {
          <span class="hljs-string">"distance"</span>: <span class="hljs-number">400</span>,
          <span class="hljs-string">"size"</span>: <span class="hljs-number">4</span>,
          <span class="hljs-string">"duration"</span>: <span class="hljs-number">0.3</span>,
          <span class="hljs-string">"opacity"</span>: <span class="hljs-number">1</span>,
          <span class="hljs-string">"speed"</span>: <span class="hljs-number">3</span>
        },
        <span class="hljs-string">"repulse"</span>: {
          <span class="hljs-string">"distance"</span>: <span class="hljs-number">200</span>,
          <span class="hljs-string">"duration"</span>: <span class="hljs-number">0.4</span>
        },
        <span class="hljs-string">"push"</span>: {
          <span class="hljs-string">"particles_nb"</span>: <span class="hljs-number">4</span>
        },
        <span class="hljs-string">"remove"</span>: {
          <span class="hljs-string">"particles_nb"</span>: <span class="hljs-number">2</span>
        }
      }
    },
    <span class="hljs-string">"retina_detect"</span>: <span class="hljs-literal">true</span>
  });
}

<span class="hljs-keyword">export</span> <span class="hljs-keyword">default</span> memo(Particles);</pre></div></div>
            
        </li>
        
    </ul>
  </div>
</body>
</html>
