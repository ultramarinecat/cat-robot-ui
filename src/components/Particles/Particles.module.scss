@charset 'utf-8';

////
//// Particles styles
////

@import 'base.scss';

.main {
  position: absolute;
  width: $full-viewport-width;
  height: $full-viewport-height;
  overflow: hidden;
  z-index: -1;
}
