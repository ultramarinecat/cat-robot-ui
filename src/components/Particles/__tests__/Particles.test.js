import React from 'react';
import { render } from 'react-testing-library';

import { Particles } from '../Particles.js';

const PARTICLES = 'particles';

describe('Particles', () => {
  beforeAll(() => {
    window.particlesJS = jest.fn();
  });

  it('should display particles effect', () => {
    const { container, getByTestId } = render(<Particles />);

    expect(getByTestId(PARTICLES)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
