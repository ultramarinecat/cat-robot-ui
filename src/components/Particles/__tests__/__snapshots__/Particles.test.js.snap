// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Particles should display particles effect 1`] = `
<div
  class="main"
  data-testid="particles"
  id="particles"
/>
`;
